File: shared/src/main/java/com/greencross/lims/data/Worklist.java
package com.greencross.lims.data;

import jsinterop.annotations.*;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@JsType(isNative=true, namespace= JsPackage.GLOBAL, name="Object")
@Setter(onMethod_= {@JsOverlay, @JsIgnore})
@Getter(onMethod_= {@JsOverlay, @JsIgnore})
@Accessors(fluent=true)
public final class Worklist {
    private String id;
    private String title;
    @JsProperty(name="created_at")
    private String created;
    private String status;
    private String remark;

    @JsOverlay
    @JsIgnore
    public Status status() {
        if (status == null) return null;
        return Status.valueOf(status);
    }

    public enum Status{
        NORMAL,         // 검증완료
        PRE_CREATE,     // Worklist 저장
        POST_CREATE,    // Worklist + Plate
        DISPOSAL,
        MERGED,
        SEQUENCING,
        ANALYZED,
        COMPLETE
    }
}

File: worklist-api/src/main/kotlin/com/greencross/lims/entity/Worklist.kt
package com.greencross.lims.entity

import com.infobip.spring.data.jdbc.annotation.processor.Schema
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.io.Serializable
import java.time.LocalDateTime
import java.util.*

@Schema("avoid")
@Table("worklist")
data class Worklist(
    @Id                                  val id:        UUID,
                                         var title:     String? = null,
    @CreatedDate @Column("create_at")    var createAt:  LocalDateTime? = null,
                                         var status:    Status? = null,
                                         var remark:    String? = null,
                                         var domain:    String? = null
) : Serializable {
    companion object {
        enum class Status{
            NORMAL, PRE_CREATE, POST_CREATE, DISPOSAL, MERGED, SEQUENCING, ANALYZED, COMPLETE
        }
    }
}